### ISSUE TITLE
Not detecting error from docker pull

### ISSUE THREAD

**Issue by dustymabe** [projectatomic/atomicapp#441]
It appears the code continues on and does not error out when `docker pull` fails. From the code below we do still get an error but it should have stopped at the `docker pull`. 

```
[root@f23 atomicapp ((c496cfc...) *%)]# atomicapp --verbose  run foo --provider docker
2015-12-06 17:41:44,472 - atomicapp.nulecule.main - DEBUG - NuleculeManager init app_path: /var/lib/atomicapp/foo-80317b5ba0d3
2015-12-06 17:41:44,472 - atomicapp.nulecule.main - DEBUG - NuleculeManager init    image: foo
2015-12-06 17:41:44,473 - atomicapp.nulecule.main - DEBUG - Request to unpack to foo to /var/lib/atomicapp/foo-80317b5ba0d3
2015-12-06 17:41:44,473 - atomicapp.nulecule.base - INFO - Unpacking image: foo to /var/lib/atomicapp/foo-80317b5ba0d3
2015-12-06 17:41:44,561 - atomicapp.nulecule.container - INFO - Pulling Docker image: foo
2015-12-06 17:41:44,561 - atomicapp.nulecule.container - DEBUG - /usr/bin/docker pull foo
Using default tag: latest
Trying to pull repository docker.io/library/foo ... not found
Error: image library/foo:latest not found
2015-12-06 17:41:45,516 - atomicapp.nulecule.container - INFO - Extracting nulecule data from image: foo to /var/lib/atomicapp/foo-80317b5ba0d3
2015-12-06 17:41:45,516 - atomicapp.nulecule.container - DEBUG - Creating docker container: /usr/bin/docker create --entrypoint /bin/true foo
Unable to find image 'foo:latest' locally
Trying to pull repository docker.io/library/foo ... not found
Error: image library/foo:latest not found
2015-12-06 17:41:46,229 - __main__ - ERROR - Command '['/usr/bin/docker', 'create', '--entrypoint', '/bin/true', 'foo']' returned non-zero exit status 1
Traceback (most recent call last):
  File "/sharedfolder/atomicapp/atomicapp/cli/main.py", line 74, in cli_run
    nm.run(**argdict)
  File "/sharedfolder/atomicapp/atomicapp/nulecule/main.py", line 184, in run
    self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
  File "/sharedfolder/atomicapp/atomicapp/nulecule/main.py", line 116, in unpack
    nodeps=nodeps, dryrun=dryrun, update=update)
  File "/sharedfolder/atomicapp/atomicapp/nulecule/base.py", line 88, in unpack
    docker_handler.extract(image, APP_ENT_PATH, dest, update)
  File "/sharedfolder/atomicapp/atomicapp/nulecule/container.py", line 81, in extract
    container_id = subprocess.check_output(run_cmd).strip()
  File "/usr/lib64/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['/usr/bin/docker', 'create', '--entrypoint', '/bin/true', 'foo']' returned non-zero exit status 1
```

**Comment by cdrage** [comment]
Resolved by #477

**Comment by containscafeine** [comment]
@cdrage @dustymabe @rtnpro, this still leaves a traceback behind.
Can I explore this more?
